generator/src/main/kotlin/io/lsdconsulting/stub/processor/typeHandlerExtensions.kt

```diff
@@ -0,0 +1,47 @@
+package io.lsdconsulting.stub.processor
+
+import javax.lang.model.element.Element
+
+fun Element.retrieveArgumentType(): String {
+    var argumentType = this.asType().toString()
+    this.asType().annotationMirrors.forEach {
+        argumentType = argumentType.replace(it.toString(), "")
+    }
+    return argumentType.replace(",", "").trim()
+}
+
+fun String.retrieveGeneric() =
+    if (this.contains("<") && this.contains(">")) {
+        this.substring(this.indexOf("<") + 1, this.lastIndexOf(">"))
+    } else if (this.contains("&lt;") && this.contains("&gt;")) {
+        this.substring(this.indexOf("&lt;") + 4, this.lastIndexOf("&gt;"))
+    } else {
+        this
+    }
+
+fun String.removeResponseEntity() =
+    if (this.contains("ResponseEntity<") && this.contains(">")) {
+        this.substring(this.indexOf("ResponseEntity<") + 15, this.lastIndexOf(">"))
+    } else if (this.contains("ResponseEntity&lt;") && this.contains("&gt;")) {
+        this.substring(this.indexOf("ResponseEntity&lt;") + 18, this.lastIndexOf("&gt;"))
+    } else {
+        this
+    }
+
+fun String.replacePrimitive() =
+    when (this) {
+        "boolean" -> "Boolean"
+        "int" -> "Integer"
+        "long" -> "Long"
+        "byte" -> "Byte"
+        "short" -> "Short"
+        "float" -> "Float"
+        "double" -> "Double"
+        "char" -> "Char"
+        else -> this
+    }
+
+fun String.retrieveResponseType(): String? {
```

Comment:
I know you've spoken before that you subscribe to the single return law, but I much prefer:
```
fun String.retrieveResponseType(): String? {
    val responseType = this.replace(Regex("\\(.*\\)"), "")
    
    if (responseType.equals("void", true)) 
        return null
    
    return responseType
}
```

Guard clauses for null suit my eye so much better than return ifs